lib/prometheus/client/metric.rb

```diff
@@ -41,21 +41,37 @@ def initialize(name,
           metric_settings: store_settings
         )
 
+        # WARNING: Our internal store can be replaced later by `with_labels`
+        # Everything we do after this point needs to still work if @store gets replaced
         init_label_set({}) if labels.empty?
       end
 
+      protected
```

Comment:
If you're only making one method `protected`, you can put it in front of `def` to avoid having to switch back to `public` after.

Reply:
Yes... A feature they added in *some* version of Ruby, but I can never find which one, because I don't know what it's called... So i'm not sure we can do that in all the versions we support. If you are sure, I can totally do that.

Reply:
Apparently it's from 2.1.
So, yeah, fair enough, 2.0 is pretty old by now